apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c

```diff
@@ -77,13 +73,425 @@ char g_query[QUERY_LENGTH];
+* @brief                To check get count of total row of cursor
+* @scenario             If used api returns DB_OK
+* @apicovered           cursor_get_count
+* @precondition         Database resource must be deinitialized
+* @postcondition        none
+*/
+void itc_arastorage_cursor_get_count_after_deinit_n(void)
+{
+	db_result_t ret;
+	const char *rel_name = "REL_COUNT";
+
+	ret = db_init();
+	TC_ASSERT_EQ("db_init", ret, DB_OK);
+
+	snprintf(g_query, QUERY_LENGTH, "CREATE RELATION %s;", rel_name);
+	ret = db_exec(g_query);
+	TC_ASSERT_EQ("db_exec", ret, DB_OK);
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "CREATE ATTRIBUTE %s DOMAIN int IN %s;", g_attribute_set[0], rel_name);
+	ret = db_exec(g_query);
+	TC_ASSERT_EQ_CLEANUP("db_exec", ret, DB_OK, db_cleanup(rel_name));
+
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "CREATE ATTRIBUTE %s DOMAIN long IN %s;", g_attribute_set[1], rel_name);
+	ret = db_exec(g_query);
+	TC_ASSERT_EQ_CLEANUP("db_exec", ret, DB_OK, db_cleanup(rel_name));
+
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "CREATE ATTRIBUTE %s DOMAIN string(32) IN %s;", g_attribute_set[2], rel_name);
+	ret = db_exec(g_query);
+	TC_ASSERT_EQ_CLEANUP("db_exec", ret, DB_OK, db_cleanup(rel_name));
+
+	memset(g_query, 0, QUERY_LENGTH);
+	sprintf(g_query, "CREATE INDEX %s.%s TYPE %s;", rel_name, g_attribute_set[0], RELATION_INDEX);
+	ret = db_exec(g_query);
+	TC_ASSERT_EQ_CLEANUP("db_exec", ret, DB_OK, db_cleanup(rel_name));
+
+	int index;
+	for (index = 0; index < DATA_SET_NUM; index++) {
+		memset(g_query, 0, QUERY_LENGTH);
+		snprintf(g_query, QUERY_LENGTH, "INSERT (%d, %ld, \'%s\') INTO %s;", g_arastorage_data_set[index].int_value,
+			g_arastorage_data_set[index].long_value, g_arastorage_data_set[index].string_value,
+			rel_name);
+		ret = db_exec(g_query);
+		TC_ASSERT_EQ_CLEANUP("db_exec", ret, DB_OK, db_cleanup(rel_name));
+	}
+
+	snprintf(g_query, QUERY_LENGTH, "SELECT %s, %s FROM %s WHERE %s > 2;", g_attribute_set[0], g_attribute_set[1], rel_name, g_attribute_set[0]);
+
+	g_cursor = NULL;
+	g_cursor = db_query(g_query);
+	TC_ASSERT_NEQ_CLEANUP("db_query", g_cursor, NULL, db_cleanup(rel_name));
+
+	ret = db_deinit();
+	TC_ASSERT_EQ_CLEANUP("db_deinit", ret, DB_OK, db_cleanup(rel_name));
+
+	cursor_row_t count;
+
+	count = cursor_get_count(g_cursor);
+	TC_ASSERT_EQ_CLEANUP("cursor_get_count", count, INVALID_CURSOR_VALUE, db_cleanup(rel_name));
+	db_cleanup(rel_name);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_arastorage_cursor_get_row_after_deinit_n
+* @brief            Get current row in cursor after deinit
+* @scenario         Get current row in cursor after db deinit
+* @apicovered       cursor_get_row
+* @precondition     Database resource must be initialized
+* @postcondition    none
+*/
+void itc_arastorage_cursor_get_row_after_deinit_n(void)
+{
+	db_result_t ret;
+	const char *rel_name = "REL_COUNT";
```

Comment:
I checked some case, all case call clean up at the last of tc.
then you don't have to define each of relation name..